services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3167,6 +3167,10 @@ private Pair<Boolean, Version> addVersion(
               }
             }
           }
+          if (Version.isPushIdTTLRePush(pushJobId) && !store.isTTLRepushEnabled()) {
```

Comment:
Emm..
This logic will only be executed in child controller, but not parent, right?

Reply:
No, it should be executed in both parent and child. Let me add another integration test to ensure that.